src/frontend/hooks/server/presets.ts

```diff
@@ -36,26 +33,3 @@ export function usePresetsMutation() {
     },
   });
 }
-
-export function useGetPresetIcon(docId: string | undefined, size: IconSize) {
-  const {projectId, projectApi} = useActiveProject();
-
-  return useQuery<string, Error>({
-    queryKey: ['presetIconFromDocId', projectId, docId, size],
-    enabled: !!docId,
-    queryFn: async () => {
-      if (!docId) {
-        throw new Error('Preset icon reference not found');
-      }
-      return await projectApi.$icons.getIconUrl(docId, {
-        mimeType: 'image/png',
-        size: size,
-        pixelDensity: 3,
-      });
-    },
-    retry: 2,
-    retryDelay: 2000,
-    staleTime: 60000,
-    placeholderData: keepPreviousData,
```

Comment:
pretty sure specifying these is not helpful. opted to remove them entirely unless there's a notable issue that can be fixed by one of them being specified (query uses defaults for some of these if not specified)

Reply:
I think we should keep the retry to 2 or even 1 (as a stop gap to the stale fastify port problem)

Reply:
React Query defaults to 3 based on the docs: https://tanstack.com/query/latest/docs/framework/react/reference/useQuery

![image](https://github.com/user-attachments/assets/a478eac9-9477-486e-a3c7-15ce71136369)

Can make it explicit if that's preferred though

Reply:
I think we should make it less than 3 because we know what the problem is, and so failing faster would be better for the UI

Reply:
unless I'm misunderstanding and you're suggesting that we should reduce the retry count (which I don't think is what you mean)

Reply:
made it 3 explicitly via https://github.com/digidem/comapeo-mobile/pull/687/commits/0c36158e7df21c5a557143e53d53b9c95fe11079

Reply:
That is what I am suggesting. Have tanstack query retry less times so that it throws an error faster. On each retry it remain in the pending state. That way the user doesn't see the loading spinner for so long and just shows the default icon when the fastify server is using a stale port

Reply:
> I think we should make it less than 3 because we know what the problem is, and so failing faster would be better for the UI

Ah i see. I'll reduce to 2 in that case. i think it's still useful to have a couple of retries because the issue doesn't always happen (but that probably largely depends on the device and other factors). I'd be curious to see how prevalent it is in a release build vs development mode

Reply:
> That way the user doesn't see the loading spinner for so long and just shows the default icon when the fastify server is using a stale port

Just to clarify, the issue not the stale port (at least in the context of this PR) but the server not being ready when resuming the app. Point understood though

Reply:
reduced to 2 via 898be8912e40b14fdc9171231b30742d50f5f75a